GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -1367,19 +1368,19 @@ void OnRevisionReadCompleted()
                         await semaphoreUpdateGrid.WaitAsync(cancellationToken);
                     }
 
-                    IEnumerable<ObjectId> headParents = null;
+                    IReadOnlyList<ObjectId>? headParents = null;
                     bool refresh = false;
                     if (!headIsHandled && ShowArtificialRevisions())
                     {
                         if (CurrentCheckout is not null)
                         {
                             // Not found, so search for its parents
-                            headParents = TryGetParents(Module, _filterInfo, CurrentCheckout);
+                            headParents = TryGetParents(Module, _filterInfo, CurrentCheckout).ToList();
                         }
 
                         // If parents are rewritten HEAD may not be included
                         // Insert the artificial commits where relevant if possible, otherwise first
-                        refresh = AddArtificialRevisions(headParents);
+                        refresh = AddArtificialRevisions(headParents ?? new List<ObjectId>());
```

Comment:
```suggestion
                        refresh = AddArtificialRevisions(headParents ?? Array.Empty<ObjectId>());
```

Why do it, if `AddArtificialRevisions` accepts null?

Reply:
Updating the comment

// HEAD where to insert the artificial was not found
// (can occur when filtering or limiting number of loaded commits)
// Try to find the revision where to attach the artificial, first is not found
// null means that no matching should be done (just add), so use an empty list if no parents were found

Reply:
Astonishingly, `Array.Empty<ObjectId>()` works without `.ToList()`.

Reply:
Will not compile

Severity	Code	Description	Project	File	Line	Suppression State	Details
Error	CS0426	The type name 'Empty<>' does not exist in the type 'Array'	GitUI	C:\dev\gc\gitextensions\GitUI\UserControls\RevisionGrid\RevisionGridControl.cs	1385	Active	

Points to System.Array as in other usages, but the disassembly has no Empty....
The only reason to change this is to not get someone try to change this later

Reply:
> Will not compile

It does and seems to work well.

Reply:
Maybe something with vs 17.8.0 preview 3?

What I get in RevisionGridControl

![image](https://github.com/gitextensions/gitextensions/assets/6248932/663aae88-1725-4b6b-844c-1f9bf7302632)

Other Array

![image](https://github.com/gitextensions/gitextensions/assets/6248932/4b933fe9-4850-4d79-aa8e-d721a78063b0)

Reply:
Today ite compiles, no changes what I am aware of...
Pushed the changes.

I would like to merge this, so the follow ups can be changed from drafts.